### PR TITLE
Use Artifacts container instead of \\cpvsbuild\drops\Roslyn share on build pipeline

### LINKED ISSUE #5265: Improve build time by reducing the number of things we copy to our build share
Relates to [AB#1087080](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1087080)

### PR DESCRIPTION
**Pull request by dmonroym**
Previously we used to publish our symbols to \\cpvsbuild\drops\Roslyn but the share got full after 3TB so moving the more modern way of uploading symbols through the Artifacts container. Now in each build under artifacts you should be able to see a folder titled "symbols"

Fixes #5265 
Partially completes some of #5905 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/project-system/pull/6310)

### CODE REVIEW

File: build/ci/official.yml [hunk 1]
```diff
@@ -50,39 +49,71 @@ steps:
...
+    sourcePath: $(Build.ArtifactStagingDirectory)/symbols
     usePat: false
     
 - task: ms-vseng.MicroBuildTasks.4305a8de-ba66-4d8b-b2d1-0dc4ecbbf5e8.MicroBuildUploadVstsDropFolder@1
   displayName: Upload VSTS Drop
   inputs:
     DropFolder: 'artifacts\$(BuildConfiguration)\VSSetup\Insertion'
-    
+
+#TODO deprecate this with PublishBuildArtifacts@1
```
Comment (dmonroym) [inline build/ci/official.yml:116]: This will be addressed via #5905

File: build/ci/official.yml [hunk 2]
```diff
@@ -50,39 +49,71 @@ steps:
   condition: succeeded()
   continueOnError: true
 
-- task: PublishTestResults@1
+- task: PublishTestResults@2
```
Comment (swesonga) [inline build/ci/official.yml:52]: >PublishTestResults@2 [](start = 8, length = 20)

Why does this need to change?

Comment (dmonroym) [reply build/ci/official.yml:52]: Version 1 is now a deprecated task

**Comment by tmeschter** [comment]
In the description, does "symbols" refer specifically to .pdbs, or does it encompass everything we want to save from a build, or something in between?

**Comment by dmonroym** [comment]
> In the description, does "symbols" refer specifically to .pdbs, or does it encompass everything we want to save from a build, or something in between?

We store everything we used to store in \\cpvsbuild.
You can see a sample of what's saved here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3854666&view=artifacts&type=publishedArtifacts

**Pull request merged**